### PR TITLE
HIVE-23413: Create a new config to skip all locks

### PR DESCRIPTION
**Pull request by pvargacl**

### CODE REVIEW

**Comment by pvargacl** [comment]
@deniskuzZ could you check this out, you already reviewed it on the review board, but I forgat about it, so now created a PR.

**Comment by pvargacl** [comment]
@deniskuzZ @pvary Do you still think we need this? If so please review, and I will check if it needs rebase.

File: ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java [hunk 1]
```diff
@@ -727,35 +751,52 @@ private void testCheckExpectedLocks(boolean sharedWrite) throws Exception {
    */
   @Test
   public void testCheckExpectedLocks2() throws Exception {
+    testCheckExpectedLocks2(true);
+  }
+  @Test
+  public void testCheckExpectedLocks2NoReadLock() throws Exception {
+    testCheckExpectedLocks2(false);
+  }
+  public void testCheckExpectedLocks2(boolean readLocks) throws Exception {
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java:760]: nit: new row

File: ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java [hunk 2]
```diff
@@ -727,35 +751,52 @@ private void testCheckExpectedLocks(boolean sharedWrite) throws Exception {
    */
   @Test
   public void testCheckExpectedLocks2() throws Exception {
+    testCheckExpectedLocks2(true);
+  }
+  @Test
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java:756]: nit: new row

**Review by pvary — APPROVED** [review]
I think this is still needed.
Skimmed through it, and only found nits.
I would be happy if @deniskuzZ could also check.
Thanks,
Peter

**Comment by deniskuzZ** [comment]
Checked, however I'm not sure if it is a good idea to give the end user the option to completely disable the locking. They could end up shooting themselves in the foot. 

From code perspective, change looks good.

**Comment by pvary** [comment]
> Checked, however I'm not sure if it is a good idea to give the end user the option to completely disable the locking. They could end up shooting themselves in the foot.

Administrator could add it to the restricted parameters which can not be set by the user, so a customer can turn off if they feel it is problematic.

**Comment by pvargacl** [comment]
@pvary So what is the conclusion? Should I add it to the restricted list by default? Or will you merge this, as is?

**Comment by pvary** [comment]
> @pvary So what is the conclusion? Should I add it to the restricted list by default? Or will you merge this, as is?

I would like to see an agreement here. This is more like a democratic process, than autocratic :)

I still think this will become useful immediately when someone has a problem with the too restrictive locking. If in doubt we might just add this to HIVE_CONF_RESTRICTED_LIST, so no rouge user can use this unless the administrator explicitly allows the usage of the new config. 

But this is only me, others might different opinions

**Pull request closed without merging**